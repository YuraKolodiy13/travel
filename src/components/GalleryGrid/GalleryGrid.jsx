import React from "react";
import {Gallery, Item} from "react-photoswipe-gallery";
import './GalleryGrid.scss'
import 'photoswipe/dist/photoswipe.css'
import 'photoswipe/dist/default-skin/default-skin.css'

const GalleryGrid = ({images}) => {

  return (
    <div className='GalleryGrid'>
      <Gallery>
        <div className="grid-gallery">
          {images?.map((item, i) =>
            <Item
              key={i}
              original={item}
              thumbnail={item}
              width='1024'
              height='683'
            >
              {({ ref, open }) => (
                <div className="destination" ref={ref} onClick={open} style={{backgroundImage: `url(${item})`}}>
                  <div className="overlay"/>
                  {i === 4 && <span>+{images.length - i}</span>}
                </div>
              )}
            </Item>
          )}
        </div>
      </Gallery>
    </div>
  )
};

export default GalleryGrid;