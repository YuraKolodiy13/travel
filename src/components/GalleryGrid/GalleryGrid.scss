.GalleryGrid{
  .grid-gallery {
    display: grid;
    grid-auto-rows: 138px;
    grid-gap: 8px;
    grid-auto-flow: dense;
    grid-template-columns: repeat(4,1fr);
    padding-bottom: 16px;
    border-bottom: 1px solid #E5E6E4;
    padding-right: 24px;
    .destination {
      position: relative;
      background-size: cover;
      background-position: center;
      display: none;
      cursor: pointer;
      &:first-of-type{
        grid-column: span 2;
        grid-row: span 2;;
      }
      &:nth-child(-n+5){
        display: block;
      }
      span{
        position: absolute;
        left: 0;
        right: 0;
        top: 0;
        bottom: 0;
        background: rgba(0, 0, 0, 0.5);
        pointer-events: none;
        font-weight: 500;
        font-size: 32px;
        line-height: 32px;
        color: #FFFFFF;
        display: flex;
        justify-content: center;
        align-items: center;
      }

      &:hover{
        .overlay {
          opacity: .8;
        }
      }

      .overlay {
        opacity: 0;
        position: absolute;
        top: 0;
        height: 100%;
        width: 100%;
        background: #0000009c;
        display: flex;
        justify-content: space-between;
        align-items: flex-end;
        transition: opacity 0.3s;
        padding: 10px;
        box-sizing: border-box;
      }
    }
  }
  .pswp img{
    max-width: 100%;
    object-fit: contain;
  }
}