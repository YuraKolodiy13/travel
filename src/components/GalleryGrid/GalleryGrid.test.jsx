import {render} from "@testing-library/react";
import React from "react";
import GalleryGrid from "./GalleryGrid";

describe('gallery component', () => {

  const images = [
    'https://img4.farvater.travel/mapkey/34015/0?size=original',
    'https://img4.farvater.travel/hotelimages/64fd1575-598c-4c39-8c46-f5621ff1682f?size=original',
    'https://img4.farvater.travel/hotelimages/5b66631d-eeea-4e30-9e25-fdfe9d55cbcd?size=original',
    'https://img4.farvater.travel/hotelimages/34981905-fa0e-47b9-b310-2d75ef189382?size=original',
    'https://img4.farvater.travel/hotelimages/ec935f6a-1293-4360-b703-08a6797c3e1a?size=original',
    'https://img4.farvater.travel/hotelimages/f25926b0-25c6-423d-8922-fe5b77782a56?size=original',
    'https://img4.farvater.travel/hotelimages/8a310410-b21b-4c6b-94f2-4766237fefc9?size=original',
    'https://img4.farvater.travel/hotelimages/1f0db15b-1d45-48da-b608-c503d345edc1?size=original',
    'https://img4.farvater.travel/hotelimages/6bea2ca4-fd03-4759-a17c-269c08a3c59a?size=original',
    'https://img4.farvater.travel/hotelimages/2a6d08f9-3a83-43ed-993c-f4656aa0e4f2?size=original',
    'https://img4.farvater.travel/hotelimages/a585642f-ae0e-4dd4-8dc9-9b301c716ab4?size=original',
    'https://img4.farvater.travel/hotelimages/15bf0363-03dd-482c-8653-70fbe8755d19?size=original',
    'https://img4.farvater.travel/hotelimages/c66e2382-a840-43fd-b343-aa57c79d6ffd?size=original',
    'https://img4.farvater.travel/hotelimages/d8bdfea6-d434-445f-8e9f-35a3950a6dd1?size=original',
    'https://img4.farvater.travel/hotelimages/b1c0a457-a4ba-4799-9082-9f65bbafc190?size=original',
    'https://img4.farvater.travel/hotelimages/27eed90a-bf32-4b5e-9192-0f48f127fb40?size=original',
    'https://img4.farvater.travel/hotelimages/f085ad4e-9902-48fb-ab6d-312775bbdf50?size=original',
    'https://img4.farvater.travel/hotelimages/d34a1a92-169d-449f-8ac1-2bc4d0d87d5d?size=original',
    'https://img4.farvater.travel/hotelimages/da1aeb77-36a7-461f-8dbe-6a8c05261404?size=original',
    'https://img4.farvater.travel/hotelimages/c48b1a8a-aa09-43e0-96a4-420557247f8e?size=original',
    'https://img4.farvater.travel/hotelimages/dba2c657-2e9a-4712-8f4d-16021341206e?size=original',
    'https://img4.farvater.travel/hotelimages/e8a431b9-21e4-4637-9632-00d317a95047?size=original',
    'https://img4.farvater.travel/hotelimages/10b43a98-e525-4895-8fcd-c45e801b1434?size=original',
    'https://img4.farvater.travel/hotelimages/5ba8abd9-3a68-492e-ab85-444bd66327cb?size=original',
    'https://img4.farvater.travel/hotelimages/75d37a57-028e-4717-9f3a-c6bd33a94564?size=original'
  ];

  test('gallery snapshot', () => {
    const gallery = render(
      <GalleryGrid images={images} />
    );
    expect(gallery).toMatchSnapshot();
  });
})