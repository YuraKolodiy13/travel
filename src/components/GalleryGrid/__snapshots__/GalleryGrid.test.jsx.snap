// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gallery component gallery snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="GalleryGrid"
      >
        <div
          class="grid-gallery"
        >
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/mapkey/34015/0?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/64fd1575-598c-4c39-8c46-f5621ff1682f?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/5b66631d-eeea-4e30-9e25-fdfe9d55cbcd?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/34981905-fa0e-47b9-b310-2d75ef189382?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/ec935f6a-1293-4360-b703-08a6797c3e1a?size=original);"
          >
            <div
              class="overlay"
            />
            <span>
              +
              21
            </span>
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/f25926b0-25c6-423d-8922-fe5b77782a56?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/8a310410-b21b-4c6b-94f2-4766237fefc9?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/1f0db15b-1d45-48da-b608-c503d345edc1?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/6bea2ca4-fd03-4759-a17c-269c08a3c59a?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/2a6d08f9-3a83-43ed-993c-f4656aa0e4f2?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/a585642f-ae0e-4dd4-8dc9-9b301c716ab4?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/15bf0363-03dd-482c-8653-70fbe8755d19?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/c66e2382-a840-43fd-b343-aa57c79d6ffd?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/d8bdfea6-d434-445f-8e9f-35a3950a6dd1?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/b1c0a457-a4ba-4799-9082-9f65bbafc190?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/27eed90a-bf32-4b5e-9192-0f48f127fb40?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/f085ad4e-9902-48fb-ab6d-312775bbdf50?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/d34a1a92-169d-449f-8ac1-2bc4d0d87d5d?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/da1aeb77-36a7-461f-8dbe-6a8c05261404?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/c48b1a8a-aa09-43e0-96a4-420557247f8e?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/dba2c657-2e9a-4712-8f4d-16021341206e?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/e8a431b9-21e4-4637-9632-00d317a95047?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/10b43a98-e525-4895-8fcd-c45e801b1434?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/5ba8abd9-3a68-492e-ab85-444bd66327cb?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
          <div
            class="destination"
            style="background-image: url(https://img4.farvater.travel/hotelimages/75d37a57-028e-4717-9f3a-c6bd33a94564?size=original);"
          >
            <div
              class="overlay"
            />
          </div>
        </div>
        <div
          aria-hidden="true"
          class="pswp"
          role="dialog"
          tabindex="-1"
        >
          <div
            class="pswp__bg"
          />
          <div
            class="pswp__scroll-wrap"
          >
            <div
              class="pswp__container"
            >
              <div
                class="pswp__item"
              />
              <div
                class="pswp__item"
              />
              <div
                class="pswp__item"
              />
            </div>
            <div
              class="pswp__ui pswp__ui--hidden"
            >
              <div
                class="pswp__top-bar"
              >
                <div
                  class="pswp__counter"
                />
                <button
                  class="pswp__button pswp__button--close"
                  title="Close (Esc)"
                  type="button"
                />
                <button
                  class="pswp__button pswp__button--share"
                  title="Share"
                  type="button"
                />
                <button
                  class="pswp__button pswp__button--fs"
                  title="Toggle fullscreen"
                  type="button"
                />
                <button
                  class="pswp__button pswp__button--zoom"
                  title="Zoom in/out"
                  type="button"
                />
                <div
                  class="pswp__preloader"
                >
                  <div
                    class="pswp__preloader__icn"
                  >
                    <div
                      class="pswp__preloader__cut"
                    >
                      <div
                        class="pswp__preloader__donut"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap"
              >
                <div
                  class="pswp__share-tooltip"
                />
              </div>
              <button
                class="pswp__button pswp__button--arrow--left"
                title="Previous (arrow left)"
                type="button"
              />
              <button
                class="pswp__button pswp__button--arrow--right"
                title="Next (arrow right)"
                type="button"
              />
              <div
                class="pswp__caption"
              >
                <div
                  class="pswp__caption__center"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="GalleryGrid"
    >
      <div
        class="grid-gallery"
      >
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/mapkey/34015/0?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/64fd1575-598c-4c39-8c46-f5621ff1682f?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/5b66631d-eeea-4e30-9e25-fdfe9d55cbcd?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/34981905-fa0e-47b9-b310-2d75ef189382?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/ec935f6a-1293-4360-b703-08a6797c3e1a?size=original);"
        >
          <div
            class="overlay"
          />
          <span>
            +
            21
          </span>
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/f25926b0-25c6-423d-8922-fe5b77782a56?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/8a310410-b21b-4c6b-94f2-4766237fefc9?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/1f0db15b-1d45-48da-b608-c503d345edc1?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/6bea2ca4-fd03-4759-a17c-269c08a3c59a?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/2a6d08f9-3a83-43ed-993c-f4656aa0e4f2?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/a585642f-ae0e-4dd4-8dc9-9b301c716ab4?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/15bf0363-03dd-482c-8653-70fbe8755d19?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/c66e2382-a840-43fd-b343-aa57c79d6ffd?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/d8bdfea6-d434-445f-8e9f-35a3950a6dd1?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/b1c0a457-a4ba-4799-9082-9f65bbafc190?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/27eed90a-bf32-4b5e-9192-0f48f127fb40?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/f085ad4e-9902-48fb-ab6d-312775bbdf50?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/d34a1a92-169d-449f-8ac1-2bc4d0d87d5d?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/da1aeb77-36a7-461f-8dbe-6a8c05261404?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/c48b1a8a-aa09-43e0-96a4-420557247f8e?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/dba2c657-2e9a-4712-8f4d-16021341206e?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/e8a431b9-21e4-4637-9632-00d317a95047?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/10b43a98-e525-4895-8fcd-c45e801b1434?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/5ba8abd9-3a68-492e-ab85-444bd66327cb?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
        <div
          class="destination"
          style="background-image: url(https://img4.farvater.travel/hotelimages/75d37a57-028e-4717-9f3a-c6bd33a94564?size=original);"
        >
          <div
            class="overlay"
          />
        </div>
      </div>
      <div
        aria-hidden="true"
        class="pswp"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="pswp__bg"
        />
        <div
          class="pswp__scroll-wrap"
        >
          <div
            class="pswp__container"
          >
            <div
              class="pswp__item"
            />
            <div
              class="pswp__item"
            />
            <div
              class="pswp__item"
            />
          </div>
          <div
            class="pswp__ui pswp__ui--hidden"
          >
            <div
              class="pswp__top-bar"
            >
              <div
                class="pswp__counter"
              />
              <button
                class="pswp__button pswp__button--close"
                title="Close (Esc)"
                type="button"
              />
              <button
                class="pswp__button pswp__button--share"
                title="Share"
                type="button"
              />
              <button
                class="pswp__button pswp__button--fs"
                title="Toggle fullscreen"
                type="button"
              />
              <button
                class="pswp__button pswp__button--zoom"
                title="Zoom in/out"
                type="button"
              />
              <div
                class="pswp__preloader"
              >
                <div
                  class="pswp__preloader__icn"
                >
                  <div
                    class="pswp__preloader__cut"
                  >
                    <div
                      class="pswp__preloader__donut"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap"
            >
              <div
                class="pswp__share-tooltip"
              />
            </div>
            <button
              class="pswp__button pswp__button--arrow--left"
              title="Previous (arrow left)"
              type="button"
            />
            <button
              class="pswp__button pswp__button--arrow--right"
              title="Next (arrow right)"
              type="button"
            />
            <div
              class="pswp__caption"
            >
              <div
                class="pswp__caption__center"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
